Sources/AppcuesKit/Presentation/Actions/ActionRegistry.swift

```diff
@@ -99,4 +87,18 @@ internal class ActionRegistry {
             return transformingAction.transformQueue(currentQueue, index: indexInCurrent, inContext: appcues)
         }
     }
+
+    private func execute(_ models: [ExperienceAction], completion: (() -> Void)? = nil) {
+        var models = models
+
+        guard let appcues = appcues, !models.isEmpty else {
+            completion?()
+            return
+        }
+
+        let next = models.removeFirst()
+        next.execute(inContext: appcues) {
+            self.execute(models, completion: completion)
+        }
+    }
```

Comment:
Not sure if it matters, but I think there is a subtle change here from how it was? Previously actions from, say, 3 different buttons would all be enqueued sequentially and executed one at a time. I think now we'd get 3 parallel `execute` calls going at once? Like I said, not sure if it matters, or even which is preferable.

I think if we wanted to keep the old behaviour, we could maybe add the completion block as a pseudo-action in the queue after the actions it enqueued so it would get called in place:

```
- BUTTON1.ACTION1
- BUTTON1.ACTION2
- BUTTON1.COMPLETION
- BUTTON2.ACTION1
- BUTTON2.COMPLETION
```

Reply:
Ah good catch. Yeah I think that sort of approach would work fine to keep the previous behavior. I can look into that before moving this forward.

Reply:
The downside of the previous approach is one bad action (ie one that doesn't call its completion handler) can forever gum up the queue 🤷‍♂️

Reply:
There is a similar implementation on Android, using an Action queue built on a Kotlin Channel. We'll need to invent some way to get a completion notification back from there as well. Going to see if a similar idea can work for both.

Reply:
@mmaatttt went ahead and reverted back the previous queue handling for the interactive actions case, so everything functions the same as before for actions triggered by user tap, etc. The only new thing here now is the `enqueue` with completion, which calls `execute` recursively. This one is only used for the new pre-step container present actions.

This matches how things were [handled on Android](https://github.com/appcues/appcues-android-sdk/pull/271) as well, and in general, I like the concept of all interactive actions flowing through a global sequential queue.

As found on Android, it doesn't work well to include the new "navigate" actions into the queue and try to await completion, since the execution of those actions actually occurs within a currently executing Action (the continue to next step typically) - and ends up deadlocking the step transition, as that continue action is never able to fully complete if something inside of it is waiting.